Ardupilog.m

```diff
@@ -427,41 +419,47 @@ function delete(obj)
             slice.numMsgs = 0;
             
             % Loop through the LogMsgGroups, slicing each one
-            for msgGroup = slice.logMsgGroups
-                % If the LogMsgGroup has been deleted, skip it
-                if ~isvalid(msgGroup)
-                    continue
+            logProps = properties(obj);
+            for i = 1:length(logProps)
+                propertyName = logProps{i}; % Get the name of the property under examination
+                % We are interested only in LogMsgGroup objects, skip the rest of the properties
+                if ~isa(obj.(propertyName),'LogMsgGroup') 
+                    continue;
                 end
                 
                 % Slice the LogMsgGroup
-                lmg_slice = slice.(msgGroup.Name).getSlice(slice_values, slice_type);
+                lmg_slice = slice.(propertyName).getSlice(slice_values, slice_type);
                 % If the slice is not empty, add it to the Ardupilog slice
                 if isempty(lmg_slice)
-                    delete(msgGroup)
+                    delete(slice.(propertyName))
```

Comment:
This doesn't have the same effect. Previously, the empty group names were removed from the slice. Now they still exist, but attempting to access them results in a message: "handle to deleted LogMsgGroup"

Example:
l = Ardupilog('41.BIN');
m = l.getSlice([0 1e3], 'LineNo');
m.RATE

In the old version, "m.RATE" shows:
No appropriate method, property, or field 'RATE' for class 'Ardupilog'.

In this new version, "m.RATE" shows:
ans = 

  handle to deleted LogMsgGroup

Note also just viewing m by typing "m" does not have the RATE property in the old version, but shows it as a [1x1 LogMsgGroup] in the new version (despite it actually being a handle to deleted object)